django/core/checks/security/base.py

```diff
@@ -182,3 +187,9 @@ def check_xframe_deny(app_configs, **kwargs):
         settings.X_FRAME_OPTIONS == 'DENY'
     )
     return [] if passed_check else [W019]
+
+
+@register(Tags.security, deploy=True)
+def check_allowed_hosts(app_configs, **kwargs):
+    passed_check = not settings.ALLOWED_HOSTS
```

Comment:
My main point was that we should check for allowed hosts being not empty, for for being empty.

PEP8 suggests using this pattern with `IF` conditionals because there it's obvious that boolean truthiness is used; in this case there is no `IF` so `bool()` makes it more explicit that boolean value is being used. I saw at least a couple of similar examples in django, e.g. one of them:

core/management/base.py:        return not bool(self.option_list)

OTOH I agree that using bool() is not terribly needed here.